web/pgadmin/static/js/helpers/ModalProvider.jsx

```diff
@@ -127,16 +127,23 @@ function confirmDelete(title, text, onDeleteClick, onCancelClick, deleteLabel =
 
 export default function ModalProvider({ children }) {
   const [modals, setModals] = React.useState([]);
-
   const showModal = (title, content, modalOptions) => {
     let id = getEpoch().toString() + crypto.getRandomValues(new Uint8Array(4));
-    setModals((prev) => [...prev, {
-      id: id,
-      title: title,
-      content: content,
-      ...modalOptions,
-    }]);
+    if(modalOptions?.modalId){
+      id = modalOptions.modalId;
+    }
+    setModals((prev) => {
+      if(prev?.find(modal=> modal.id === modalOptions?.modalId)){
+        return [...prev];
```

Comment:
return prev instead of array destructure?

Reply:
Done.